From 983077806ca4a19aaf5f41022d9bb1f27baa83c0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Per=20=C3=98yvind=20Karlsen=20=28proyvind=29?=
 <peroyvind@mandriva.org>
Date: Sun, 30 Nov 2014 11:20:42 +0100
Subject: [PATCH] try fix auto generation of perl Makefile...

---
 mythtv/Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/mythtv/Makefile b/mythtv/Makefile
index 54d4403..731e110 100644
--- a/mythtv/Makefile
+++ b/mythtv/Makefile
@@ -31,6 +31,7 @@ all_no_test: all
 force:
 
 config.mak:
+	rm -f bindings/perl/Makefile
 	$(error run configure to create $@)
 
 # Override PWD in case this is run from a makefile at a higher level with
@@ -45,6 +46,9 @@ libs: external
 filters: libs
 programs: libs filters
 
+bindings/perl/Makefile: config.mak
+	cd $(shell dirname $@); \
+	perl Makefile.PL INSTALLDIRS=vendor
 
 subdirs: $(SUBDIRS)
 
-- 
1.9.0

