diff -up mythtv-31.0/mythtv/libs/libmythui/mythmainwindow.cpp.6~ mythtv-31.0/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-31.0/mythtv/libs/libmythui/mythmainwindow.cpp.6~	2020-03-26 01:34:55.776899369 +0100
+++ mythtv-31.0/mythtv/libs/libmythui/mythmainwindow.cpp	2020-03-26 01:35:28.850036058 +0100
@@ -2769,10 +2769,10 @@ void MythMainWindow::StartLIRC(void)
     if (!QFile::exists(config_file))
         config_file = QDir::homePath() + "/.lircrc";
 
-    /* lircd socket moved from /dev/ to /var/run/lirc/ in lirc 0.8.6 */
+    /* lircd socket moved from /dev/ to /run/lirc/ in lirc 0.8.6 */
     QString lirc_socket = "/dev/lircd";
     if (!QFile::exists(lirc_socket))
-        lirc_socket = "/var/run/lirc/lircd";
+        lirc_socket = "/run/lirc/lircd";
 
     d->m_lircThread = new LIRC(
         this,
diff -up mythtv-31.0/mythtv/programs/mythfrontend/globalsettings.cpp.6~ mythtv-31.0/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-31.0/mythtv/programs/mythfrontend/globalsettings.cpp.6~	2020-03-23 15:32:28.000000000 +0100
+++ mythtv-31.0/mythtv/programs/mythfrontend/globalsettings.cpp	2020-03-26 01:34:55.777899373 +0100
@@ -2024,7 +2024,7 @@ static HostTextEditSetting *LircDaemonDe
     QString lirc_socket = "/dev/lircd";
 
     if (!QFile::exists(lirc_socket))
-        lirc_socket = "/var/run/lirc/lircd";
+        lirc_socket = "/run/lirc/lircd";
 
     ge->setValue(lirc_socket);
 
