From ec069ed675f816be3168a7638f4700979bfb288b Mon Sep 17 00:00:00 2001
From: Colin Guthrie <colin@mageia.org>
Date: Sun, 22 Apr 2012 02:37:38 +0100
Subject: [PATCH 103/103] Fix dns-sd detection

---
 mythtv/configure |   11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git mythtv/configure mythtv/configure
index b1d930f..a1d8f67 100755
--- mythtv/configure
+++ mythtv/configure
@@ -4162,7 +4162,16 @@ fi
 enabled libcrypto && check_lib openssl/rsa.h RSA_new -lcrypto || disable libcrypto
 
 if test $target_os != darwin ; then
-    enabled libdns_sd && check_lib dns_sd.h DNSServiceRegister -ldns_sd || disable libdns_sd
+    if enabled libdns_sd; then
+        check_lib dns_sd.h DNSServiceRegister -ldns_sd || disable libdns_sd
+        if disabled libdns_sd && pkg-config --exists avahi-compat-libdns_sd ; then
+            enable libdns_sd
+            dns_sd_flags=`pkg-config --cflags avahi-compat-libdns_sd`
+            add_cflags $dns_sd_flags
+            add_cxxflags $dns_sd_flags
+            check_lib dns_sd.h DNSServiceRegister `pkg-config --libs avahi-compat-libdns_sd` || disable libdns_sd
+        fi
+    fi
 fi
 
 if enabled libxml2 ; then
-- 
1.7.10

