From f146c3eace076b2e6a7737c4f6d39932ac15fe3d Mon Sep 17 00:00:00 2001
From: Colin Guthrie <colin@mageia.org>
Date: Sun, 22 Apr 2012 02:37:38 +0100
Subject: [PATCH 103/105] Fix dns-sd detection

---
 mythtv/configure | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/mythtv/configure b/mythtv/configure
index 8551480..a7da12f 100755
--- a/mythtv/configure
+++ b/mythtv/configure
@@ -6608,7 +6608,7 @@
 enabled libcrypto && check_lib crypto openssl/rsa.h RSA_new -lcrypto || disable libcrypto
 
 if test $target_os != darwin ; then
-    enabled libdns_sd && check_lib dns_sd dns_sd.h DNSServiceRegister -ldns_sd || disable libdns_sd
+    enabled libdns_sd && check_lib dns_sd avahi-compat-libdns_sd/dns_sd.h DNSServiceRegister -ldns_sd || disable libdns_sd
 fi
 
 if enabled libxml2 ; then
diff --git a/mythtv/libs/libmythbase/bonjourregister.h b/mythtv/libs/libmythbase/bonjourregister.h
index 8551480..a7da12f 100644
--- a/mythtv/libs/libmythbase/bonjourregister.h
+++ b/mythtv/libs/libmythbase/bonjourregister.h
@@ -3,7 +3,7 @@
 
 #include <QObject>
 #include <QMutex>
-#include <dns_sd.h>
+#include <avahi-compat-libdns_sd/dns_sd.h>
 #include "mythbaseexp.h"
 
 class QSocketNotifier;
-- 
1.8.4

