From f063b19facd153c2b2844e9643d570b9466b4b6f Mon Sep 17 00:00:00 2001
From: Stuart Auchterlonie <stuarta@squashedfrog.net>
Date: Fri, 2 Jan 2015 11:52:58 +0000
Subject: [PATCH] Add missing QObject include. Should resolve Qt5 build error

---
 mythtv/libs/libmyth/programtypes.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mythtv/libs/libmyth/programtypes.cpp b/mythtv/libs/libmyth/programtypes.cpp
index a277034..e2fe498 100644
--- a/mythtv/libs/libmyth/programtypes.cpp
+++ b/mythtv/libs/libmyth/programtypes.cpp
@@ -2,6 +2,7 @@
 
 #include <QDateTime>
 #include <QMutex>
+#include <QObject>
 
 #include "programtypes.h"
 #include "mythdate.h"
