diff -uraN mythtv-32.0/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp omv-mythtv-32.0/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp
--- mythtv-32.0/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2022-02-23 18:01:37.000000000 +0100
+++ omv-mythtv-32.0/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2022-12-12 20:29:54.034338169 +0100
@@ -15,6 +15,7 @@
 
 
 #include <algorithm>
+#include <array>
 #include <fstream>
 #include <iostream>
 #include <cstdlib>
