diff --git a/mythtv/i18n/mythfrontend_de.qm b/mythtv/i18n/mythfrontend_de.qm
index 46551bd..f1ddc6f 100644
Binary files a/mythtv/i18n/mythfrontend_de.qm and b/mythtv/i18n/mythfrontend_de.qm differ
diff --git a/mythtv/i18n/mythfrontend_de.ts b/mythtv/i18n/mythfrontend_de.ts
index 4abd787..fec2c27 100644
--- a/mythtv/i18n/mythfrontend_de.ts
+++ b/mythtv/i18n/mythfrontend_de.ts
@@ -4391,7 +4391,7 @@ Gerät unterstützt bis zu %1</translation>
         <location filename="../programs/mythfrontend/customedit.cpp" line="466"/>
         <source>Category type (%1)</source>
         <comment>List of hardcoded category types</comment>
-        <translation>Kategorietyp (%!)</translation>
+        <translation>Kategorietyp (%1)</translation>
     </message>
     <message>
         <location filename="../programs/mythfrontend/customedit.cpp" line="507"/>
diff --git a/mythtv/libs/libmythtv/eithelper.cpp b/mythtv/libs/libmythtv/eithelper.cpp
index b0e493c..5b66988 100644
--- a/mythtv/libs/libmythtv/eithelper.cpp
+++ b/mythtv/libs/libmythtv/eithelper.cpp
@@ -1083,6 +1083,7 @@ static void init_fixup(FixupMap &fix)
 
     for (int i = 2037; i <= 2057; ++i)
        fix[ (long long)i << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2059LL << 32 | 2U << 16] = EITFixUp::kFixUK;
     fix[ 2061LL << 32 | 2U << 16] = EITFixUp::kFixUK;
     fix[ 2063LL << 32 | 2U << 16] = EITFixUp::kFixUK;
     fix[ 2064LL << 32 | 2U << 16] = EITFixUp::kFixUK;
@@ -1091,7 +1092,11 @@ static void init_fixup(FixupMap &fix)
     fix[ 2069LL << 32 | 2U << 16] = EITFixUp::kFixUK;
     fix[ 2076LL << 32 | 2U << 16] = EITFixUp::kFixUK;
     fix[ 2081LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2089LL << 32 | 2U << 16] = EITFixUp::kFixUK;
     fix[ 2096LL << 32 | 2U << 16] = EITFixUp::kFixUK | EITFixUp::kFixHTML;
+    fix[ 2107LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2112LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2116LL << 32 | 2U << 16] = EITFixUp::kFixUK;
     fix[ 2301LL << 32 | 2U << 16] = EITFixUp::kFixUK | EITFixUp::kFixHTML;
     fix[ 2302LL << 32 | 2U << 16] = EITFixUp::kFixUK;
     fix[ 2303LL << 32 | 2U << 16] = EITFixUp::kFixUK;
