diff -Nru a/mythtv/libs/libmythui/libmythui.pro b/mythtv/libs/libmythui/libmythui.pro
--- a/mythtv/libs/libmythui/libmythui.pro	2019-01-14 12:53:51.000000000 +0100
+++ b/mythtv/libs/libmythui/libmythui.pro	2019-01-14 18:36:17.932334135 +0100
@@ -18,6 +18,7 @@
 INCLUDEPATH += ../.. ../../external/FFmpeg
 
 LIBS += -L../libmythbase -lmythbase-$$LIBVERSION
+LIBS += -L../../external/FFmpeg/libavcodec -lmythavcodec
 
 QMAKE_CLEAN += $(TARGET) $(TARGETA) $(TARGETD) $(TARGET0) $(TARGET1) $(TARGET2)
 
